src/tv2-common/__tests__/transitionSettings.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { AtemTransitionSettings } from '../../../../tv-automation-state-timeline-resolver/packages/timeline-state-resolver-types/src'
+import { PartType } from '../../tv2-constants'
+import { AtemSourceIndex } from '../../types/atem'
+import { TV2BlueprintConfig, TV2BlueprintConfigBase, TV2StudioConfigBase } from '../blueprintConfig'
+import { PartDefinition, PartTransition } from '../inewsConversion'
+import { TransitionSettings } from '../transitionSettings'
+
+const DURATION: number = 50
+
+const WIPE_STYLE: string = 'WIPE'
+const MIX_STYLE: string = 'MIX'
+const DIP_STYLE: string = 'DIP'
+
+describe('transitionsSettingsSuite', () => {
+	let mockConfig: TV2BlueprintConfig
+
+	beforeEach(() => {
+		mockConfig = ({} as any) as TV2BlueprintConfig
+	})
+
+	describe('TransitionSettings', () => {
+		it('return empty when no transition', () => {
+			const partDefinition: PartDefinition = createPartDefinition()
```

Comment:
I would suggest keeping it there.
We should only remove the type declaration when it's abundantly clear what the right side of the expresssion evaluate to. (In this case it's pretty clear, so you could argue that we could remove it using the just stated argument. Personally, I do prefer having them there since it makes it easier to jump to the interface)

 

> and adds extra work if the type at some point changes

If the return type ever changes then the contract of the method changes and that SHOULD give us work to do where it is used because the method is now supposed to be used in a different way. I would be extremely worried if I could change the contract of a method and there would be zero compiler errors.

In this specific case, if the return type of the method changes, then the purpose of the unit test most likely also changes, so we should go and reevaluate if the unit test is still correct.

Sidenote, the idea of TypeScript is that our code is strongly typed. If we are able to just change return types with no consequences, then our code isn't strongly type.

Reply:
I agree that we shouldn't be able to change a contract without compiler errors, but I will argue that the primary changes shouldn't be local variable type re-assignments. It should rather be from the usage of those local variables, e.g. accessing properties or invoking function calls.

A comment to the sidenote: Even without the type restriction for the local variable, the code is still strongly typed, just not explicitly typed.